network-api/networkapi/mozfest/models.py

```diff
@@ -50,7 +51,11 @@ class MozfestPrimaryPage(FoundationMetadataPageMixin, Page):
 
     def get_context(self, request):
         context = super().get_context(request)
-        return get_page_tree_information(self, context)
+        context = get_page_tree_information(self, context)
+        # Also make sure that these pages always tap into the mozfest newsletter for the footer!
+        mozfest_footer = Signup.objects.filter(name__iexact='mozfest').first()
```

Comment:
What if there are no results?

Reply:
(This is looking at the Signup snippet, correct? It seems like the review app doesn't have any snippets, but the submission still "works". I just can't see where it's going)

Reply:
The React conversion [in main.js](https://github.com/mozilla/foundation.mozilla.org/blob/mozfest-footer-update/source/js/main.js#L249-L265) looks at the HTML generated in https://github.com/mozilla/foundation.mozilla.org/pull/3169/files#diff-a7829fada9ac08105a722faacb866cefR13 and will see "there is no id here, I will default to id=0"

Reply:
This is then further corrected for in the signup submission process: if a valid submission comes in (e.g. the csrf token is correct) but it's for an unknown signup snippet, [the code will use the default signup](https://github.com/mozilla/foundation.mozilla.org/blob/mozfest-footer-update/network-api/networkapi/campaign/views.py#L54-L67) by building a throw-away instance so it can get the default newsletter we've set in our Signup model.

Reply:
So if `mozfest` doesn't exist in the database, it falls back to the MozillaFestival newsletter signup?

Reply:
correct

Reply:
I think my concern is more around a future-us testing this on a review app or locally and wondering why it's doing the wrong thing. As noted in Slack, let's put this into the migration path.

Reply:
migration added